runtime/android/core/src/org/xwalk/core/XWalkCoreWrapper.java

```diff
@@ -244,6 +271,24 @@ private boolean findSharedCore() {
         return true;
     }
 
+    private boolean findDownloadedCore() {
+        String libDir = mWrapperContext.getDir(PRIVATE_LIB_DIR, Context.MODE_PRIVATE).toString();
+        String dexPath = libDir + File.separator + "classes.dex";
+        String dexOutputPath = mWrapperContext.getDir("dex", 0).getAbsolutePath();
+        ClassLoader localClassLoader = ClassLoader.getSystemClassLoader();
+        mBridgeLoader = new DexClassLoader(dexPath, dexOutputPath, libDir, localClassLoader);
+
+        if (!checkCoreVersion() || !checkCoreArchitecture()) {
+            mBridgeLoader = null;
+            mCoreStatus = XWalkLibraryInterface.STATUS_INCOMPLETE_LIBRARY;
```

Comment:
It's incorrect to set the status here. The error status has already been defined by `checkCoreVersion` or `checkCoreArchitecture`. This will overwrite the real status.